charts/skypilot/templates/api-deployment.yaml

```diff
@@ -0,0 +1,191 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: skypilot-api-server
+  namespace: {{ .Release.namespace }}
+spec:
+  replicas: {{ .Values.apiService.replicas }}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: skypilot-api
+  template:
+    metadata:
+      labels:
+        app: skypilot-api
+    spec:
+      automountServiceAccountToken: {{ .Values.kubernetesCredentials.useApiServerCluster }}
+      serviceAccountName: skypilot-api-sa
+      containers:
+      - name: skypilot-api
+        image: {{ .Values.apiService.image }}
+        imagePullPolicy: Always
+        resources:
```

Comment:
`request` is crucial in production environment to avoid over-subscription. Maybe we can expose the entire `resources` field to helm values, so that users can tune `request` and other resources (ephemeral-storage, for example) based on their requirements.

Reply:
Good idea - refactored into a general `resources` field.